From: Victor Seva <linuxmaniac@torreviejawireless.org>
Date: Thu, 22 Jun 2023 17:29:48 +0200
Subject: [PATCH] tls: fix build for openssl < 1.1.1

OPENSSL_INIT_ATFORK was introduced in libssl 1.1.1
error introduced at 9d6bfb96528c49e6aaa39aa47be877ca528c3537

(cherry picked from commit 82f5fcbf88ee3058bd9da520b528c86393cc422a)
---
 src/modules/tls/tls_init.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/modules/tls/tls_init.c b/src/modules/tls/tls_init.c
index 8071ec3..82a850c 100644
--- a/src/modules/tls/tls_init.c
+++ b/src/modules/tls/tls_init.c
@@ -740,7 +740,11 @@ int tls_h_mod_pre_init_f(void)
 	LM_DBG("preparing tls env for modules initialization\n");
 #if OPENSSL_VERSION_NUMBER >= 0x010100000L && !defined(LIBRESSL_VERSION_NUMBER)
 	LM_DBG("preparing tls env for modules initialization (libssl >=1.1)\n");
+#if OPENSSL_VERSION_NUMBER >= 0x010101000L
 	OPENSSL_init_ssl(OPENSSL_INIT_ATFORK, NULL);
+#else
+	OPENSSL_init_ssl(0, NULL);
+#endif
 #else
 	LM_DBG("preparing tls env for modules initialization (libssl <=1.0)\n");
 	SSL_library_init();
